examples/adaptive_library/BPMTriangularRotorNotches.py

```diff
@@ -0,0 +1,246 @@
+"""
+.. _ref_BPM_Triangular_Rotor_Notches:
+
+Triangular Rotor Notches for IPM
+================================
+Adaptive Template script to create triangular rotor notches to improve NVH performance.
+"""
+# %%
+# .. note::
+#    For more information on the use of Adaptive Templates in Motor-CAD, and how to create, modify
+#    and debug Adaptive Templates Scripts, see :ref:`ref_adaptive_templates_UG` in the
+#    :ref:`ref_user_guide`.
+#
+# This script is designed to be run from Motor-CAD template "e9". If no Motor-CAD file is open, the
+# e9 template will be loaded.
+#
+# This script uses the following adaptive parameters:
+#
+# * Notch Angle (-4)
+#
+# * Notch Sweep (5)
+#
+# * Notch Depth (1)
+#
+# * Notches per Pole (2)
+#
+# If these parameters are not already set up in the Motor-CAD file, the parameters will be
+# automatically set, with the default values shown in brackets.
+#
+# To set an adaptive geometry for a Motor-CAD file, a script must be loaded in to the Adaptive
+# Templates tab (Geometry -> Editor -> Adaptive Templates) in Motor-CAD and run. When the option
+# 'Geometry Templates Type' is set to 'Adaptive', this script is automatically run repeatedly to
+# keep the Adaptive Geometry set in Motor-CAD.
+#
+# .. image:: ../../images/Adaptive_Templates_GUI_Screenshot.png
+#
+# This Python script can also be executed externally. When executed externally, a Motor-CAD instance
+# will be launched and a file based on the "e9" template will be saved to a temporary folder. This
+# script will be loaded into the Adaptive Templates tab.
+
+# %%
+# Perform Required imports
+# ------------------------
+# Import pymotorcad to access Motor-CAD. Import triangular_notch to create the notch geometry region
+# with Adaptive Template geometry. Import os, tempfile and shutil to open and save a temporary .mot
+# file if none is open.
+import os
+import shutil
+import sys
+import tempfile
+
+import ansys.motorcad.core as pymotorcad
+from ansys.motorcad.core.geometry_shapes import triangular_notch
+from ansys.motorcad.core.rpc_client_core import _is_running_in_internal_scripting
+
+# %%
+# Connect to Motor-CAD
+# --------------------
+# If this script is loaded into the Adaptive Templates file in Motor-CAD, the current Motor-CAD
+# instance will be used.
+#
+# If the script is run externally: a new Motor-CAD instance will be opened, the e9 IPM motor
+# template will be loaded and the file will be saved to a temporary folder.
+# To keep a new Motor-CAD instance open after executing the script, use the option
+# ``mc = pymotorcad.MotorCAD(reuse_parallel_instances=True)`` when opening the new instance.
```

Comment:
@jgsdavies is this the recommended way to get the Motor-CAD instance to stay open after the script is complete?